##**Урок 5. Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование**

In [7]:
import numpy as np

**1.Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, а объем выборки n = 256.**

In [1]:
M = 80  
n = 256  
sigma=16  
alpha=0.05 

$$z_{\alpha/2} = 1.96$$

In [2]:
z_alpfa_2=1.96

In [5]:
se=sigma/np.sqrt(n)
se

1.0

In [8]:
t_stat(M, se, n, alpha, 'two-sided')

(78.03072611034499, 81.96927388965501)

Ответ: Доверительный интервал [ 78.04 ; 81.96 ] с вероятностью 95%

**2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.**

In [9]:
X = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
n = len(X)  # Объём выборки
# X_cp = np.sum(X)/n  # Среднее арифмитическое для данной выборки
X_cp = X.mean()  # Среднее арифмитическое для данной выборки
alpha = 0.05  # Статистический уровень значимости
nu = n-1  # Число степеней свободы
t_975_9 = 2.262  # Коэффициент Стьюдента табличный t(p=0.975, nu=9)
# sigma = X.var(ddof=1)**(1/2)  # Несмещённое среднеквадратичное отклонение
sigma = X.std(ddof=1)  # Несмещённое среднеквадратичное отклонение
print(f'Объём выборки n = {n}')
print(f'Число степеней свободы nu = {nu}')
print(f'Среднее выборочное X_cp = {X_cp}')
print(f'Статистический уровень значимости alpha = {alpha}')
print(f'Коэффициент Стьюдента табличный t(p=0.975, nu=9) = {t_975_9}')
print(f'Несмещённое среднеквадратичное отклонение sigma = {sigma}')
X

Объём выборки n = 10
Число степеней свободы nu = 9
Среднее выборочное X_cp = 6.590000000000001
Статистический уровень значимости alpha = 0.05
Коэффициент Стьюдента табличный t(p=0.975, nu=9) = 2.262
Несмещённое среднеквадратичное отклонение sigma = 0.4508017549014448


array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [10]:
t_975_9*sigma/np.sqrt(10)

0.32246174408757394

In [11]:
X_cp + t_975_9*sigma/np.sqrt(10)

6.912461744087575

In [12]:
X_cp - t_975_9*sigma/np.sqrt(10)

6.267538255912426

Ответ: Доверительный интервал [ 6.27 ; 6.91 ] с вероятностью 95%

**3.Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.**


$H_0 \quad \mu = \mu_0$

$H_1 \quad \mu > \mu_0$

$Z_H = \frac{\overline{X} - \mu_0}{\sigma} \cdot \sqrt{n}$

$\alpha = 0.05$

$Z_H = \frac{17.5 - 17}{2} \cdot \sqrt{100} = 2.5$

$Z = 1.654$

$Z_H > Z \quad \text{верна гипотеза }H_1$

Ответ:  верна  гипотеза H1

**4.Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?**

In [13]:

X = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
n = len(X)  # Объём выборки
X_0 = 200  # Среднее арифмитическое генеральной совокупности
# X_cp = np.sum(X)/n  # Среднее арифмитическое для данной выборки
X_cp = X.mean()  # Среднее арифмитическое для данной выборки
alpha = 0.01  # Статистический уровень значимости
nu = n-1  # Число степеней свободы
t_t = 3.25  # Коэффициент Стьюдента табличный t(p=0.995, nu=9)
# sigma = X.var(ddof=1)**(1/2)  # Несмещённое среднеквадратичное отклонение
sigma = X.std(ddof=1)  # Несмещённое среднеквадратичное отклонение
t_n = (X_cp - X_0)/(sigma/np.sqrt(n))
print(f'Объём выборки n = {n}')
print(f'Число степеней свободы nu = {nu}')
print(f'Среднее арифмитическое генеральной совокупности X_0 = {X_0}')
print(f'Среднее выборочное X_cp = {X_cp}')
print(f'Статистический уровень значимости alpha = {alpha}')
# print(f'Коэффициент Стьюдента табличный t(p=0.995, nu=9) = {t_995_9}')
print(f'Несмещённое среднеквадратичное отклонение sigma = {sigma}')
print(f't табличное при alpha=5% t_t = {t_t}')
print(f't расчётное при alpha=5% t_n = {t_n}')
X

Объём выборки n = 10
Число степеней свободы nu = 9
Среднее арифмитическое генеральной совокупности X_0 = 200
Среднее выборочное X_cp = 198.5
Статистический уровень значимости alpha = 0.01
Несмещённое среднеквадратичное отклонение sigma = 4.453463071962462
t табличное при alpha=5% t_t = 3.25
t расчётное при alpha=5% t_n = -1.0651074037450896


array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])


$\mu = \frac{202 + 203 + 199 + 197 + 195 + 201 + 200 + 204 + 194 + 190}{10} = 198.5$

$S = \sqrt{ \frac{(202 - 198.5)^2 + (203 - 198.5)^2 + (199 - 198.5)^2 + (197 - 198.5)^2 + (195 - 198.5)^2 + (201 - 198.5)^2 + (200 - 198.5)^2 + (204 - 198.5)^2 + (194 - 198.5)^2 + (190 - 198.5)^2}{10-1} } = 4.45346$

$H_0 \quad \mu = \mu_0$

$H_1 \quad \mu \neq \mu_0$

$\alpha = 0.01$

$T_H = \frac{198.5 - 200}{4.45346} \cdot \sqrt{10} = −1.065$

$T = \pm 3.250 $

$-3.250 < T_H < 3.250 \quad \text{верна гипотеза }H_0$

Ответ: верна нулевая гипотеза H0